src/core/lib/iomgr/cfstream_handle.h

```diff
@@ -65,6 +66,9 @@ class CFStreamHandle final {
   dispatch_queue_t dispatch_queue_;
 
   gpr_refcount refcount_;
+
+  GRPC_ALLOW_CLASS_TO_USE_NON_PUBLIC_NEW
```

Comment:
I'd actually like to make these macros go away.  I suggest just making the ctor and dtor public.  Even if you need to retain the factory function for some reason, you can just document the fact that callers are expected to use it instead of directly constructing the object.